lib/neovim-plugin.nix

```diff
@@ -89,6 +89,13 @@
           options.${namespace}.${name} =
             {
               enable = lib.mkEnableOption originalName;
+              lazyLoad = lib.nixvim.mkLazyLoadOption originalName;
+              # TODO:
+              # lazyLoadDefaults = {
+              #   # FIXME: infinite recursion
+              #   # enable = lib.mkDefault (cfg.lazyLoad.backendSettings != { });
+              # };
+              # };
```

Comment:
Looks like this is infinitely recursive because you can't use the `lazyload` option's value to define the `lazyload` option itself.

Unless I'm misinterpreting what `lazyLoadDefaults` is/does?

---

I believe the inf-rec could be avoided by working _within_ the submodule, e.g.:

```nix
types.submodule (
  { config, ... }:
  {
    # e.g. using a config def, but maybe an option default makes more sense?
    config.enable = lib.mkDefault (config.backendSettings != {});
  }
)
```